src/KeyringController.ts

```diff
@@ -1,45 +1,61 @@
-const encryptor = require('@metamask/browser-passworder');
-const HdKeyring = require('@metamask/eth-hd-keyring');
-const { normalize: normalizeAddress } = require('@metamask/eth-sig-util');
-const SimpleKeyring = require('@metamask/eth-simple-keyring');
+import type { TypedTransaction, TxData } from '@ethereumjs/tx';
+import * as encryptorUtils from '@metamask/browser-passworder';
+import HDKeyring from '@metamask/eth-hd-keyring';
+import { normalize as normalizeAddress } from '@metamask/eth-sig-util';
+import SimpleKeyring from '@metamask/eth-simple-keyring';
+import type {
+  Hex,
+  Json,
+  Bytes,
+  Keyring,
+  KeyringClass,
+  Eip1024EncryptedData,
+} from '@metamask/utils';
 // TODO: Stop using `events`, and remove the notice about this from the README
 // eslint-disable-next-line import/no-nodejs-modules
-const { EventEmitter } = require('events');
-const ObservableStore = require('obs-store');
+import { EventEmitter } from 'events';
+import ObservableStore from 'obs-store';
+
+import { KeyringType, KeyringControllerError } from './constants';
+import {
+  MessageParams,
+  SerializedKeyring,
+  KeyringControllerArgs,
+  KeyringControllerState,
+} from './types';
+import { stripHexPrefix } from './utils';
 
 const defaultKeyringBuilders = [
   keyringBuilderFactory(SimpleKeyring),
-  keyringBuilderFactory(HdKeyring),
+  keyringBuilderFactory(HDKeyring),
 ];
 
-const KEYRINGS_TYPE_MAP = {
-  HD_KEYRING: 'HD Key Tree',
-  SIMPLE_KEYRING: 'Simple Key Pair',
-};
+class KeyringController extends EventEmitter {
+  keyringBuilders: { (): Keyring<Json>; type: string }[];
 
-/**
- * Strip the hex prefix from an address, if present.
- *
- * @param {string} address - The address that might be hex prefixed.
- * @returns {string} The address without a hex prefix.
- */
-function stripHexPrefix(address) {
-  if (address.startsWith('0x')) {
-    return address.slice(2);
-  }
-  return address;
-}
+  public store: typeof ObservableStore;
```

Comment:
Isn't ObservableStore a class? It seems like we don't need to use `typeof` here.

Reply:
I'll resolve this conversation since there must be something here I don't quite understand, so this is fine.